src/datagen/adequate_datagen.js

```diff
@@ -0,0 +1,226 @@
+            key: identifier,
+            format: schema.format,
+            supportedFormats: 'email, uuid, uri, ipv4/ipv6',
+          },
+        };
+        logger['info'](errorInfo);
+        return '';
+      }
+    }
+  }
+
+  if (schema.pattern) {
+    try {
+      const regex = new RegExp(schema.pattern);
+      return new RandExp(regex).gen();
+    } catch (err) {
+      logger['info'](`Invalid Schema Pattern - ${schema.pattern}`);
+      logger['error'](err);
+      return '';
+    }
+  }
+
+  const low = schema.minLength || 1;
+  const high = schema.maxLength || low + 10;
+
+  const lengthOfString = getRandomNumber(low, high, {returnInteger: true});
+  return getRandomString(lengthOfString);
+}
+
+/**
+ * Generates a random array with items that complies with schema.
+ * @param {object} schema Specification of Array.
+ * @param {string} identifier Name of the Array.
+ * @return {array} Random Array.
+ */
+function getMockArray(schema, identifier) {
+  const mockArray = [];
+  const lengthOfMockArray = getRandomNumber(1, 10, {returnInteger: true});
+  for (let index = 0; index < lengthOfMockArray; index++) {
+    mockArray.push(getMockData(schema.items, identifier));
+  }
+  return mockArray;
+}
+
+/**
+ * Generates a random object that complies with schema.
+ * @param {object} schema Specification of Object.
+ * @param {string} [identifier = '$'] Name of the Object.
+ * @return {object} Random Object.
+ */
+function getMockObject(schema, identifier = '$') {
+  const mockObject = {};
+  const keys = Object.keys(schema.properties);
+  keys.forEach(function(key) {
+    const keySchema = schema.properties[key];
+    mockObject[key] =
+      getMockData(keySchema, identifier + '.' + key);
+  });
+  return mockObject;
+}
+
+/**
+ * Generates a random data that complies with schema.<br>
+ * Use case: generating random requestbody which complies with schema.
+ * @param {object} schema Specification of data.
+ * @param {string} [identifier = '$'] Name of the Data.
+ * @return {object} Random Object.
+ */
+function getMockData(schema, identifier = '$') {
+  if (!schema) return undefined;
+
+  if (schema.oneOf) {
+    const schemas = schema.oneOf;
+    return getMockData(
+        schemas[Math.floor(Math.random() * schemas.length)], identifier);
+  }
+
+  if (schema.enum) {
+    const items = schema.enum;
+    return items[Math.floor(Math.random() * items.length)];
+  }
+
+  switch (schema.type) {
+    case DataType.BOOLEAN:
+      return [true, false][Math.floor(Math.random() * 2)];
+    case DataType.INTEGER:
+      return getMockInteger(schema);
+    case DataType.NUMBER:
+      return getMockNumber(schema);
+    case DataType.STRING:
+      return getMockString(schema, identifier);
+    case DataType.ARRAY:
+      return getMockArray(schema, identifier);
+    case DataType.OBJECT:
+      return getMockObject(schema, identifier);
+    default:
+      logger['error'](`No support for ${schema.type}`);
+      return '';
```

Comment:
Return `null` or `undefined` maybe?

Reply:
@siriscac 
As we discussed today on our sync call, we will document things we support and ask the user of the app to take a look. So ideally, we shouldn't be hitting this line of code. But yeah, at the worse if we hit this code, we will send a null .